Integration test `test_storage_s3/test.py::test_seekable_formats_url` is flaky

Test fails with timeout:



<details>
<summary>Details</summary>

```
2024-06-04 06:04:52 [ 527 ] DEBUG : Executing query DROP TABLE IF EXISTS mt_test_alter_with_merge_work_1717480978 SYNC on node1 (cluster.py:3449, query)
__________________________ test_seekable_formats_url ___________________________
[gw0] linux -- Python 3.10.12 /usr/bin/python3

started_cluster = <helpers.cluster.ClickHouseCluster object at 0x7f50a70ef760>

    def test_seekable_formats_url(started_cluster):
        bucket = started_cluster.minio_bucket
        instance = started_cluster.instances["dummy"]  # type: ClickHouseInstance
    
        table_function = f"s3(s3_parquet, structure='a Int32, b String', format='Parquet')"
>       exec_query_with_retry(
            instance,
            f"insert into table function {table_function} SELECT number, randomString(100) FROM numbers(1500000) settings s3_truncate_on_insert=1",
        )

test_storage_s3/test.py:1169: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
helpers/test_tools.py:147: in exec_query_with_retry
    res = instance.query(query, timeout=30, settings=settings)
helpers/cluster.py:3450: in query
    return self.client.query(
helpers/client.py:36: in wrap
    return func(self, *args, **kwargs)
helpers/client.py:74: in query
    ).get_answer()
helpers/client.py:221: in get_answer
    self.process.wait(timeout=DEFAULT_QUERY_TIMEOUT)
/usr/lib/python3.10/subprocess.py:1209: in wait
    return self._wait(timeout=timeout)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Popen: returncode: None args: ['/clickhouse', 'client', '--host', '172.16.5...>
timeout = 600

    def _wait(self, timeout):
        """Internal implementation of wait() on POSIX."""
        if self.returncode is not None:
            return self.returncode
    
        if timeout is not None:
            endtime = _time() + timeout
            # Enter a busy loop if we have a timeout.  This busy loop was
            # cribbed from Lib/threading.py in Thread.wait() at r71065.
            delay = 0.0005 # 500 us -> initial delay of 1 ms
            while True:
                if self._waitpid_lock.acquire(False):
                    try:
                        if self.returncode is not None:
                            break  # Another thread waited.
                        (pid, sts) = self._try_wait(os.WNOHANG)
                        assert pid == self.pid or pid == 0
                        if pid == self.pid:
                            self._handle_exitstatus(sts)
                            break
                    finally:
                        self._waitpid_lock.release()
                remaining = self._remaining_time(endtime)
                if remaining <= 0:
                    raise TimeoutExpired(self.args, timeout)
                delay = min(delay * 2, remaining, .05)
>               time.sleep(delay)
E               Failed: Timeout >900.0s


```

</details>


[cidb](https://play.clickhouse.com/play?user=play#CldJVEgKICAgIGZhbHNlIGFzIG9ubHlfbWFzdGVyClNFTEVDVCAqCkZST00gKApTRUxFQ1QKICAgIHB1bGxfcmVxdWVzdF9udW1iZXIgYXMgcHJudW0sCiAgICBzdWJzdHIoY29tbWl0X3NoYSwgMSw5KSBhcyBzaGEsCiAgICByb3VuZCh0ZXN0X2R1cmF0aW9uX21zIC8gMTAwMCwgMikgYXMgZHVyLAogICAgY2hlY2tfc3RhcnRfdGltZSwKICAgIGZvcm1hdFJlYWRhYmxlVGltZURlbHRhKGludERpdihub3coKSAtIGNoZWNrX3N0YXJ0X3RpbWUsIDM2MDApICogMzYwMCkgYXMgYWdvLAogICAgY2hlY2tfbmFtZSwKICAgIHRlc3RfbmFtZSwKICAgIHJlcG9ydF91cmwKRlJPTSBjaGVja3MKV0hFUkUgMQogICAgQU5EIGNoZWNrX3N0YXJ0X3RpbWUgPj0gbm93KCkgLSBJTlRFUlZBTCA3IERBWQogICAgQU5EIChwdWxsX3JlcXVlc3RfbnVtYmVyID0gMCBPUiBub3Qgb25seV9tYXN0ZXIpCiAgICBBTkQgdGVzdF9zdGF0dXMgIT0gJ1NLSVBQRUQnCiAgICBBTkQgdGVzdF9zdGF0dXMgTElLRSAnRiUnCiAgICBBTkQgY2hlY2tfc3RhdHVzICE9ICdzdWNjZXNzJwopCldIRVJFIHRlc3RfbmFtZSBsaWtlICcldGVzdF9zdG9yYWdlX3MzL3Rlc3QucHk6OnRlc3Rfc2Vla2FibGVfZm9ybWF0c191cmwlJwpPUkRFUiBCWSBjaGVja19zdGFydF90aW1lIERFU0MsIGNoZWNrX25hbWUsIHRlc3RfbmFtZQ==)

<details>
<summary>Details</summary>


| prnum | sha | dur | check_start_time | ago | check_name | test_name | report_url |
|-:|:-|-:|-:|:-|:-|:-|:-|
| 64952 | 05ea297c0 | 900 | 2024-06-07 04:34:55 | 3 days and 5 hours | Integration tests (tsan) [5/6] | test_storage_s3/test.py::test_seekable_formats_url | https://s3.amazonaws.com/clickhouse-test-reports/64952/05ea297c0d79f55f65ded88ec5284fa9e8c34753/integration_tests__tsan__[5_6].html |
| 64924 | 1881107f0 | 900 | 2024-06-06 19:45:45 | 3 days and 13 hours | Integration tests (tsan) [5/6] | test_storage_s3/test.py::test_seekable_formats_url | https://s3.amazonaws.com/clickhouse-test-reports/64924/1881107f0773d4aa0e195ef5cd6bb8f8a17b8cc3/integration_tests__tsan__[5_6].html |
| 0 | eb724ad8c | 900 | 2024-06-06 14:11:44 | 3 days and 19 hours | Integration tests (tsan) [5/6] | test_storage_s3/test.py::test_seekable_formats_url | https://s3.amazonaws.com/clickhouse-test-reports/0/eb724ad8c7ebf0ad99e1ef6711d967d090e82109/integration_tests__tsan__[5_6].html |
| 64798 | 7ce67265c | 900 | 2024-06-05 17:51:48 | 4 days and 15 hours | Integration tests (tsan) [5/6] | test_storage_s3/test.py::test_seekable_formats_url | https://s3.amazonaws.com/clickhouse-test-reports/64798/7ce67265c70075009617c1b43700d351bf43a9ad/integration_tests__tsan__[5_6].html |
| 64775 | 1565aec14 | 900 | 2024-06-04 06:26:41 | 6 days and 3 hours | Integration tests (tsan) [5/6] | test_storage_s3/test.py::test_seekable_formats_url | https://s3.amazonaws.com/clickhouse-test-reports/64775/1565aec14281de2069fc18dbe6594cea8e813e66/integration_tests__tsan__[5_6].html |
| 0 | b6e3b9c78 | 900 | 2024-06-04 04:45:48 | 6 days and 4 hours | Integration tests (tsan) [5/6] | test_storage_s3/test.py::test_seekable_formats_url | https://s3.amazonaws.com/clickhouse-test-reports/0/b6e3b9c7847b635b1a49cb22d56f2b5c0a6061b8/integration_tests__tsan__[5_6].html |
| 64768 | e9465fc79 | 900 | 2024-06-04 04:13:41 | 6 days and 5 hours | Integration tests (tsan) [5/6] | test_storage_s3/test.py::test_seekable_formats_url | https://s3.amazonaws.com/clickhouse-test-reports/64768/e9465fc7907846eb043826de253fdd62cbaf288d/integration_tests__tsan__[5_6].html |




</details>

--- a/tests/integration/test_storage_s3/test.py
+++ b/tests/integration/test_storage_s3/test.py
@@ -1148,11 +1148,12 @@ def test_url_reconnect_in_the_middle(started_cluster):
         assert result == "1000000\t3914219105369203805\n"
 
 
-
 # At the time of writing the actual read bytes are respectively 148 and 169, so -10% to not be flaky
-@pytest.mark.parametrize("format_name,expected_bytes_read", [("Parquet", 133), ("ORC", 150)])
+@pytest.mark.parametrize(
+    "format_name,expected_bytes_read", [("Parquet", 133), ("ORC", 150)]
+)
 def test_seekable_formats(started_cluster, format_name, expected_bytes_read):
-    expected_lines=1500000
+    expected_lines = 1500000
     instance = started_cluster.instances["dummy"]  # type: ClickHouseInstance
 
     table_function = f"s3(s3_{format_name.lower()}, structure='a Int32, b String', format='{format_name}')"
@@ -1185,10 +1186,10 @@ def test_seekable_formats(started_cluster, format_name, expected_bytes_read):
 @pytest.mark.parametrize("format_name", ["Parquet", "ORC"])
 def test_seekable_formats_url(started_cluster, format_name):
     bucket = started_cluster.minio_bucket
-    expected_lines=1500000
+    expected_lines = 1500000
     instance = started_cluster.instances["dummy"]  # type: ClickHouseInstance
 
-    format_name_lower=format_name.lower()
+    format_name_lower = format_name.lower()
     table_function = f"s3(s3_{format_name_lower}, structure='a Int32, b String', format='{format_name}')"
     exec_query_with_retry(
         instance,

--- a/tests/integration/test_storage_s3/test.py
+++ b/tests/integration/test_storage_s3/test.py
@@ -1148,31 +1148,27 @@ def test_url_reconnect_in_the_middle(started_cluster):
         assert result == "1000000\t3914219105369203805\n"
 
 
-def test_seekable_formats(started_cluster):
-    bucket = started_cluster.minio_bucket
+
+# At the time of writing the actual read bytes are respectively 148 and 169, so -10% to not be flaky
+@pytest.mark.parametrize("format_name,expected_bytes_read", [("Parquet", 133), ("ORC", 150)])
+def test_seekable_formats(started_cluster, format_name, expected_bytes_read):
+    expected_lines=1500000
     instance = started_cluster.instances["dummy"]  # type: ClickHouseInstance
 
-    table_function = f"s3(s3_parquet, structure='a Int32, b String', format='Parquet')"
+    table_function = f"s3(s3_{format_name.lower()}, structure='a Int32, b String', format='{format_name}')"
     exec_query_with_retry(
         instance,
-        f"insert into table function {table_function} SELECT number, randomString(100) FROM numbers(1000000) settings s3_truncate_on_insert=1",
-        timeout=100,
+        f"INSERT INTO TABLE FUNCTION {table_function} SELECT number, randomString(100) FROM numbers({expected_lines}) settings s3_truncate_on_insert=1",
+        timeout=300,
     )
 
     result = instance.query(f"SELECT count() FROM {table_function}")
-    assert int(result) == 1000000
-
-    table_function = f"s3(s3_orc, structure='a Int32, b String', format='ORC')"
-    exec_query_with_retry(
-        instance,
-        f"insert into table function {table_function} SELECT number, randomString(100) FROM numbers(1500000) settings s3_truncate_on_insert=1",
-        timeout=100,
-    )
+    assert int(result) == expected_lines
 
     result = instance.query(
         f"SELECT count() FROM {table_function} SETTINGS max_memory_usage='60M', max_download_threads=1"
     )
-    assert int(result) == 1500000
+    assert int(result) == expected_lines
 
     instance.query(f"SELECT * FROM {table_function} FORMAT Null")
 
@@ -1183,35 +1179,31 @@ def test_seekable_formats(started_cluster):
     result = result.strip()
     assert result.endswith("MiB")
     result = result[: result.index(".")]
-    assert int(result) > 150
+    assert int(result) > 140
 
 
-def test_seekable_formats_url(started_cluster):
+@pytest.mark.parametrize("format_name", ["Parquet", "ORC"])
+def test_seekable_formats_url(started_cluster, format_name):
     bucket = started_cluster.minio_bucket
+    expected_lines=1500000
     instance = started_cluster.instances["dummy"]  # type: ClickHouseInstance
 
-    table_function = f"s3(s3_parquet, structure='a Int32, b String', format='Parquet')"
+    format_name_lower=format_name.lower()
+    table_function = f"s3(s3_{format_name_lower}, structure='a Int32, b String', format='{format_name}')"
     exec_query_with_retry(
         instance,
-        f"insert into table function {table_function} SELECT number, randomString(100) FROM numbers(1500000) settings s3_truncate_on_insert=1",
-        timeout=100,
+        f"INSERT INTO TABLE FUNCTION {table_function} SELECT number, randomString(100) FROM numbers({expected_lines}) settings s3_truncate_on_insert=1",
+        timeout=300,
     )
 
     result = instance.query(f"SELECT count() FROM {table_function}")
-    assert int(result) == 1500000
+    assert int(result) == expected_lines
 
-    table_function = f"s3(s3_orc, structure='a Int32, b String', format='ORC')"
-    exec_query_with_retry(
-        instance,
-        f"insert into table function {table_function} SELECT number, randomString(100) FROM numbers(1500000) settings s3_truncate_on_insert=1",
-        timeout=100,
-    )
-
-    table_function = f"url('http://{started_cluster.minio_host}:{started_cluster.minio_port}/{bucket}/test_parquet', 'Parquet', 'a Int32, b String')"
+    url_function = f"url('http://{started_cluster.minio_host}:{started_cluster.minio_port}/{bucket}/test_{format_name_lower}', '{format_name}', 'a Int32, b String')"
     result = instance.query(
-        f"SELECT count() FROM {table_function} SETTINGS max_memory_usage='60M'"
+        f"SELECT count() FROM {url_function} SETTINGS max_memory_usage='60M'"
     )
-    assert int(result) == 1500000
+    assert int(result) == expected_lines
 
 
 def test_empty_file(started_cluster):

